mono/metadata/icall.c

```diff
@@ -6892,34 +6892,85 @@ ves_icall_System_Environment_GetCommandLineArgs (MonoError *error)
 }
 
 #ifndef HOST_WIN32
-static MonoArray *
-mono_icall_get_environment_variable_names (MonoError *error)
+
+static void
+mono_new_string_utf8_to_array (MonoArrayHandle array, gsize index,
```

Comment:
Why having this function in this file while `mono_new_string_utf16_to_array` is in `object.c`? How about you try to keep it consistent: either both functions in `object.c` or both functions here.

Reply:
I think it was because of ifdef win32, because of this being the only use. But yeah, both in object.c not bad idea.

Reply:
That is, in a larger change, there was maybe more uses of the one function & only one use of the other. Anyway, I can put them both in object.c.